src/gsNurbs/gsTensorBSplineBasis.h

```diff
@@ -371,8 +371,10 @@ class gsTensorBSplineBasis : public gsTensorBasis<d,T>
 
         for (short_t dim = 0; dim < d; ++dim)
         {
-            Self_t::component(dim).knots().supportIndex_into(ti[dim], tmp_vec);
-            result.row(dim) = tmp_vec.row(0);
+            const gsKnotVector<T> & kv = Self_t::component(dim).knots();
+            kv.supportIndex_into(ti[dim], tmp_vec);
+            result.row(dim).noalias() =
+                tmp_vec.cwiseMax(0).cwiseMin(kv.numElements());
```

Comment:
The reason for the cwise trick is that supportIndex_into can now (due to the handling of ghosts) sometimes return things below zero?

Reply:
Yes, exactly.